src/js/shop/components/Checkout/Payment.js

```diff
@@ -0,0 +1,218 @@
+import React, { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import orderBy from "lodash.orderby";
+import { FormattedMessage } from "react-intl";
+import Select from "react-select";
+import useAsync from "react-use/esm/useAsync";
+
+import { PaymentConsumer } from "../../../data/shop/payment";
+import Loader from "../../../components/loaders";
+import { ErrorMessage } from "../Info";
+import { FormField, FormGroup } from "./FormFields";
+import { BodyCart } from "../Cart";
+
+const paymentConsumer = new PaymentConsumer();
+
+const useFetchPaymentMethods = () => {
+    const {
+        loading,
+        error,
+        value = [],
+    } = useAsync(async () => {
+        const methodList = await paymentConsumer.listMethods();
+        return methodList;
```

Comment:
```suggestion
    } = useAsync(paymentConsumer.listMethods)
```
If `paymentConsumer.listMethods` already returns a promise with the list of methods, this can be simplified.

Reply:
unfortunately `react-use/useAsync` requires the wrapper boilerplate, it's one of the things that bothers me with this hook :P